analyzers/src/SonarAnalyzer.Common/Helpers/SonarLintXmlReader.cs

```diff
@@ -32,49 +32,68 @@ public class SonarLintXmlReader
 
     private readonly SonarLintXml sonarLintXml;
 
-    private bool? ignoreHeaderCommentsCS;
-    private bool? ignoreHeaderCommentsVB;
+    public string[] Exclusions { get; }
+    public string[] Inclusions { get; }
+    public string[] GlobalExclusions { get; }
+    public string[] TestExclusions { get; }
+    public string[] TestInclusions { get; }
+    public string[] GlobalTestExclusions { get; }
+    public List<SonarLintXmlRule> ParametrizedRules { get; }
+    private bool IgnoreHeaderCommentsCS { get; }
+    private bool IgnoreHeaderCommentsVB { get; }
+    private bool AnalyzeGeneratedCodeCS { get; }
+    private bool AnalyzeGeneratedCodeVB { get; }
+
+    public SonarLintXmlReader(SourceText sonarLintXmlText)
+    {
+        sonarLintXml = ParseContent(sonarLintXmlText);
+
+        var settings = SettingsToDictionary();
```

Comment:
@pavel-mikula-sonarsource  Settings is a `List<SonarLintXmlKeyValuePair>`.
I'm aware that turning it into a Dictionary is an indirection - but the trade-off of not going over the list and setting a case for each Key to fill out the Property IMO worths it.

Ideally, IMO the dictionary should come from `SonarLintXml` object, but it's quite a change for this PR.